pkgs/applications/networking/browsers/firefox/common.nix

```diff
@@ -92,6 +92,15 @@ let
 
   browserPatches = [
     ./env_var_for_system_dir.patch
+    (fetchpatch {
```

Comment:
I do not think that these belong here. They do not apply to say `firefox-esr-60`. We are building a lot of different FireFox flavors from the same common expressions. The tor browsers also use this expression.

There is two ways I could see this work:
* Find a generic solution that works on "all" the flavors.
* Use a patch specific to the flavor. E.g. pull from the ESR repo when building the ESR version(s) etc..

I prefer the 2nd approach as that will allow us to remove that from specific versions without triggering rebuilds of all the others.

Reply:
Is there already an example for doing this?

Reply:
https://github.com/NixOS/nixpkgs/blob/cff3ab79b5af82f404392a8eb4fc501ffddf61cc/pkgs/applications/networking/browsers/firefox/packages.nix#L19-L21

and

https://github.com/NixOS/nixpkgs/blob/cff3ab79b5af82f404392a8eb4fc501ffddf61cc/pkgs/applications/networking/browsers/firefox/packages.nix#L51-L55

are two examples how we do patches per flavor.